qa/tasks/vstart_runner.py

```diff
@@ -271,8 +271,10 @@ def put_file(self, src, dst, sudo=False):
         try:
             shutil.copy(src, dst)
         except exception as e:
-            if sys.version_info.major < 3 and e.message.find('are the same '
-               'file') != -1:
+            if sys.version_info.major < 3:
+                if e.message.find('are the same file') != -1:
+                    return
+            elif 'are the same file' in str(e):
```

Comment:
why can't we catch the `SameFileException` from `shutil`?

Reply:
Please take a look at the code just above it -
```
    def put_file(self, src, dst, sudo=False):
        if sys.version_info.major < 3:
            exception = shutil.Error
        elif sys.version_info.major >= 3:
            exception = shutil.SameFileError

        try:
            shutil.copy(src, dst)
        except exception as e:
            if sys.version_info.major < 3:
                if e.message.find('are the same file') != -1:
                    return
            elif 'are the same file' in str(e):
                return
            raise e
```

https://github.com/rishabh-d-dave/ceph/blob/fs-shell-SameFileError/qa/tasks/vstart_runner.py#L266-L269

Reply:
Okay... then I'm still confused why we're parsing the exception stringification when `sys.version_info.major >= 3`.

Reply:
Right. Checking exception's message is redundant when Python version >= 3. Fixed.